support-modules/aws/src/main/scala/com/gu/aws/AwsS3Client.scala

```diff
@@ -1,34 +1,58 @@
 package com.gu.aws
 
-import java.io.InputStream
-
-import com.amazonaws.regions.Regions
-import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder, AmazonS3URI}
+import com.gu.aws.AwsS3Client.S3Location
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
 
+import java.io.InputStream
+import java.net.URI
 import scala.io.Source
 import scala.util.Try
 
 object AwsS3Client extends AwsS3Client {
 
-  private val s3: AmazonS3 =
-    AmazonS3ClientBuilder
-      .standard()
-      .withRegion(Regions.EU_WEST_1)
-      .withCredentials(CredentialsProvider)
+  private val s3: S3Client =
+    S3Client
+      .builder()
+      .region(Region.EU_WEST_1)
+      .credentialsProvider(CredentialsProvider)
       .build()
 
-  def withStream[RESULT](block: InputStream => Try[RESULT])(uri: AmazonS3URI): Try[RESULT] = for {
-    stream <- Try(s3.getObject(uri.getBucket, uri.getKey).getObjectContent)
-    result <- block(stream)
-    _ = stream.close()
-  } yield result
+  def withStream[RESULT](s3Location: S3Location)(block: InputStream => Try[RESULT]): Try[RESULT] = {
+
+    val objectRequest: GetObjectRequest = GetObjectRequest.builder
+      .key(s3Location.key)
+      .bucket(s3Location.bucket)
+      .build
+
+    for {
+      stream <- Try(s3.getObject(objectRequest))
+      result <- block(stream)
+      _ = stream.close()
+    } yield result
+  }
 
   // convenience method below
-  def fetchAsString: AmazonS3URI => Try[String] = withStream(is => Try(Source.fromInputStream(is).mkString))
+  def fetchAsString(s3Location: S3Location): Try[String] =
+    withStream(s3Location) { is =>
+      Try(Source.fromInputStream(is).mkString)
+    }
+
+  case class S3Location(bucket: String, key: String)
+  def parseUri(uri: String): S3Location = {
+    val s3Uri = s3.utilities().parseUri(URI.create(uri))
+    S3Location(
+      s3Uri.bucket().get(), // throw if not present
```

Comment:
elsewhere in this file we're returning `Try` - would that be more consistent here?

Reply:
yeah why not I'll do that.
sometimes dealing with the `Try`s can get out of hand when all you want to do is throw a 500 when it goes wrong (this function is only used for reading a config value to decide which file to load from S3)

Reply:
done - it also actually encouraged me to neaten up S3Loader.scala into a for comprehension by making that change.